agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2142,6 +2142,16 @@ sMacAddr backhaul_manager::get_sta_bssid(
     return bssid;
 }
 
+std::shared_ptr<backhaul_manager::sRadioInfo>
+backhaul_manager::get_sta_radio(const sMacAddr &sta_mac)
+{
+    auto radio = std::find_if(
+        slaves_sockets.begin(), slaves_sockets.end(), [&sta_mac](std::shared_ptr<sRadioInfo> r) {
```

Comment:
I think that parameter to predicate function should be const reference
```
const std::shared_ptr<sRadioInfo> &radio_info
```

Reply:
Good point. 
Also, since it does not share the ownership of the sRadioInfo, I don't see why it couldn't be a `sRadioInfo &` instead of a shared_ptr.